GMP-Report Comb. & pseudo Plots: -- reduce title sizes in the underlying programs

alexsav815 opened the pull request:
…ing programs ...

demeler reviewed (commented):
Aspect ratio problems appear to be solved.
Axis titles are still cut off when using a 12 pt Arial base font. They look fine with 10 pt Arial base font, so it seems that an explicit 10 pt Arial font should be used for extra long names:

![image](https://github.com/user-attachments/assets/59090c71-1d89-4e73-87d3-571ebc08a410)

We need to make sure that all installations have the basic microsoft fonts installed (times new roman, arial). All the million other fonts are not needed. The Beckman installation for example doesn't have Arial installed. This should be standardized so this ticket can rely on 10 pt Arial being there.

demeler commented:
@alexsav815 :   Regenerating a GMP report with #194 resulted in reduced sizes of fonts, but there was still a scaling issue:
From the same report, we get correctly scaled fonts:
![correct](https://github.com/user-attachments/assets/73be061a-8fc8-4ede-bd8c-1b9359e45240) 
and incorrectly (appearing squashed) scaled fonts: 
![incorrect](https://github.com/user-attachments/assets/935afc0e-8142-47cb-a353-4ef9a72528cc)
Can you handle the scaling issue?

alexsav815 commented:
Not sure what you are talking about - I just regenerated the same report on umontana server and get this:

![image](https://github.com/user-attachments/assets/bba5d47b-4c8d-4ded-a762-39206868e826)

alexsav815 commented:
It looks to me you may still have problem with using that script for pulling & rebuilding correct branch/pull request..

demeler commented:
I was able to reproduce the problem with the latest master branch - Emre just updated UMontana, this is further discussed below. But first to this plot, there are still additional issues.
I just noticed that the s-value range was set to 10s, while the analysis range went to 50s and the results had non-zero solute contributions throughout this range. This is evident when you create the plot with the same datasets manually:

![combined](https://github.com/user-attachments/assets/dfae4696-9318-46f6-9d50-17f1b1859c48)

So the plot is showing only the lower end of the s-distribution.

Secondly, the image in the PDF is squashed and its aspect ratio is distorted. See screenshots above. I just performed another run with the protocol development tool of UltraScan v. 7669 (the latest master branch, so not a pullrequest issue) and received another report with the same problem, and additional ones:

[BDemeler-041025-DNAStandard20c-14_6-v3.pdf](https://github.com/user-attachments/files/19725741/BDemeler-041025-DNAStandard20c-14_6-v3.pdf)

on Page 49, you see the same s-value range issue - it should go to 20s for this plot, and the squashing of the aspect ratio of the image. Interestingly, if you copy the image from the PDF file and paste it into GIMP the image's correct aspect ratio is restored. So you can visually compare the two images and immediately should be able to see the problem.
Not all images have this problem. However, there are still other problems. On p47, the aspect ratio appears to be correct, but the text is cut off at the left and the right edge of the plot title ("B" on the left, and "0" on the right). So, we still have three issues with rendering these plots in the pdf file:

1. s-value range in the combo plot is wrong
2. aspect ratio is *sometimes* distorted
3. text is cut off if the runID is too long

alexsav815 commented:
Borries, we have this very discussion in the pull request thread - it's not in the master branch yet, so you are not using the changes from "alexey-dev-issue445" by utilizing current master branch...

On other points - this is something new to me, was not in the design (what was in the design is to pass all s-ranges from the protocol, both global and for individual plots & apply them to plots). 

So, nothing is wrong in the current implementation. If you need changes to the design, there will be new development.

demeler commented:
We definitely need to display the full s-range. The combine distributions
program handles the s-range automatically upon loading the model, so I am
surprised this is not happening here. The x-axis range should be autoscale,
so it shouldn't be necessary to pass any range to this program, the module
generating the s-value overlays should automatically scale to the largest
range.

On Sun, Apr 13, 2025 at 12:28 PM alexsav815 ***@***.***>
wrote:

> Borries, we have this very discussion in the pull request thread - it's
> not in the master branch yet, so you are not using the changes from
> "alexey-dev-issue445" by utilizing current master branch...
>
> On other points - this is something new to me, was not in the design (what
> was in the design is to pass all s-ranges from the protocol, both global
> and for individual plots & apply them to plots).
>
> So, nothing is wrong in the current implementation. If you need changes to
> the design, there will be new development.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ehb54/ultrascan3/pull/194#issuecomment-2800066529>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB2N727OXAB7ACNEYNLFWXD2ZKUD7AVCNFSM6AAAAABZ6CHAXOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQMBQGA3DMNJSHE>
> .
> You are receiving this because your review was requested.Message ID:
> ***@***.***>
> *alexsav815* left a comment (ehb54/ultrascan3#194)
> <https://github.com/ehb54/ultrascan3/pull/194#issuecomment-2800066529>
>
> Borries, we have this very discussion in the pull request thread - it's
> not in the master branch yet, so you are not using the changes from
> "alexey-dev-issue445" by utilizing current master branch...
>
> On other points - this is something new to me, was not in the design (what
> was in the design is to pass all s-ranges from the protocol, both global
> and for individual plots & apply them to plots).
>
> So, nothing is wrong in the current implementation. If you need changes to
> the design, there will be new development.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ehb54/ultrascan3/pull/194#issuecomment-2800066529>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB2N727OXAB7ACNEYNLFWXD2ZKUD7AVCNFSM6AAAAABZ6CHAXOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQMBQGA3DMNJSHE>
> .
> You are receiving this because your review was requested.Message ID:
> ***@***.***>
>

alexsav815 commented:
Borries, I think I addressed all points - aspect ratios, font, and s-ranges for combined plots (overall, not individual).
Attached are 2 re-gerenerated GMP reports for the recent runs on UMontana server. 

 
[BDemeler-041025-DNAStandard20c-14_6-v3.pdf](https://github.com/user-attachments/files/19765870/BDemeler-041025-DNAStandard20c-14_6-v3.pdf)
[032125_Au-NP-v4.pdf](https://github.com/user-attachments/files/19765871/032125_Au-NP-v4.pdf)

(NOTE: again, if you want to try it, you need to pull&rebuild 'alexey-dev-issue445' branch, or this very pull request - the master does NOT have these changes..)

Questions: 
1. Since I changed the way how s-ranges are treated for the overall comb. plots, do we need the 'Global Parameters for Combined Plots' section in the Experiment setting? (see pics below)

![image](https://github.com/user-attachments/assets/82891c45-14c9-494b-b2b6-22b5a91f9515)

![image](https://github.com/user-attachments/assets/119af899-4c58-4595-bb57-3e20949b579b)
 
2. Do we treat differently individual comb. plots vs. overall comb. plots differently - do we still want to use s-ranges for individual channels in 2DSA settings, or use whatever is the biggest range detected  for the models? Currently, I read s-ranges for channels from AProfile's  2DSA  settings...

demeler reviewed (approved):
All reports look correct.

demeler reviewed (approved):
Addresses needs by Alexander.
